tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -0,0 +1,281 @@
+                                endpoint_url=S3_CFG["s3_url"])
+
+        self.log.info("Step 1: Perform write Operations :")
+        self.dtm_obj.perform_write_op(bucket_prefix=f"bucket-{test_prefix}",
+                                      object_prefix=f"object-{test_prefix}",
+                                      no_of_clients=self.test_cfg['clients'],
+                                      no_of_samples=self.test_cfg['samples'],
+                                      obj_size=self.test_cfg['size'],
+                                      log_file_prefix=test_prefix, queue=wr_output)
+        resp = wr_output.get()
+        assert_utils.assert_true(resp[0], resp[1])
+
+        buckets = s3_test_obj.bucket_list()[1]
+        self.log.info("Step 1: Successfully created %s buckets & performed WRITEs with variable "
+                      "size objects.", len(buckets))
+
+        output = Queue()
+        self.log.info("Step 2: Start Continuous DELETEs of buckets %s in background", buckets)
+        args = {'test_prefix': test_prefix, 'test_dir_path': self.test_dir_path,
+                'skipput': True, 'skipget': True, 'bkt_list': buckets, 'output': output}
+
+        thread = threading.Thread(target=self.ha_obj.put_get_delete,
+                                  args=(event, s3_test_obj,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        self.log.info("Step 2: Successfully started DELETEs in background")
+
+        event.set()
+        self.log.info("Step 3: Perform Single m0d Process Restart During Delete Operations")
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj,
+                                            pod_prefix=POD_NAME_PREFIX,
+                                            container_prefix=MOTR_CONTAINER_PREFIX,
+                                            process=self.rgw_process, check_proc_state=False)
+        assert_utils.assert_true(resp, "Failure observed during process restart/recovery")
+        event.clear()
+        self.log.info("Step 3: Successfully Performed Single m0d Process Restart During Delete "
+                      "Operations")
+
+        self.log.info("Step 4: Verify status for In-flight DELETEs while service was restarting")
+        thread.join()
+        del_resp = tuple()
+        while len(del_resp) != 2:
+            del_resp = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        event_del_bkt = del_resp[0]
+        fail_del_bkt = del_resp[1]
+        assert_utils.assert_false(len(fail_del_bkt),
+                                  f"Bucket deletion failed before/after restart:{fail_del_bkt}")
+
+        if event_del_bkt:
+            self.log.info("Retrying bucket deletion for buckets: %s", event_del_bkt)
+            resp = s3_test_obj.delete_multiple_buckets(bucket_list=event_del_bkt)
```

Comment:
Please let me know why this retry is needed.

Reply:
This is to check, once service is restarted, delete works on retrying.